pkg/virt-controller/watch/vm/vm_test.go

```diff
@@ -2576,6 +2576,18 @@ var _ = Describe("VirtualMachine", func() {
 			Expect(vmi1.Spec.Domain.Firmware.UUID).NotTo(Equal(vmi3.Spec.Domain.Firmware.UUID))
 		})
 
+		It("should have different firmware UUIDs when using diffrent namespace", func() {
+			vm1, _ := watchtesting.DefaultVirtualMachineWithNames(true, "testvm1", "testvmi1")
+			vmi1 := controller.setupVMIFromVM(vm1)
+
+			// intentionally use the same names
+			vm2, _ := watchtesting.DefaultVirtualMachineWithNames(true, "testvm1", "testvmi1")
+			vm2.Namespace = "other-namespace"
+			vmi2 := controller.setupVMIFromVM(vm2)
+
```

Comment:
I'd add:
`Expect(vm1.Name).To(Equal(vmi2.Name))`
`Expect(vmi1.Namespace).ToNot(Equal(vmi2.Namespace))`

Reply:
Thanks, Done !